caddytls/challenge_provider.go

```diff
@@ -0,0 +1,6 @@
+package caddytls
+
+import "github.com/xenolf/lego/acme"
+
+// ChallengeProvider type to be used in Caddy plugins over acme.ChallengeProvider directly, to avoid https://github.com/mattfarina/golang-broken-vendor
+type ChallengeProvider acme.ChallengeProvider
```

Comment:
Instead of creating a new file, put this in tls.go, like right after or before DNSProviderConstructor is defined.